clients/unshaded/src/main/java/tachyon/client/lineage/TachyonLineageFileSystem.java

```diff
@@ -0,0 +1,172 @@
+/*
+ * Licensed to the University of California, Berkeley under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package tachyon.client.lineage;
+
+import java.io.IOException;
+import java.util.List;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.base.Preconditions;
+
+import tachyon.Constants;
+import tachyon.TachyonURI;
+import tachyon.annotation.PublicApi;
+import tachyon.client.file.FileOutStream;
+import tachyon.client.file.TachyonFileSystem;
+import tachyon.client.file.options.OutStreamOptions;
+import tachyon.exception.TachyonException;
+import tachyon.job.CommandLineJob;
+import tachyon.job.Job;
+import tachyon.thrift.FileDoesNotExistException;
+import tachyon.thrift.LineageDeletionException;
+import tachyon.thrift.LineageDoesNotExistException;
+import tachyon.thrift.LineageInfo;
+
+/**
+ * Tachyon lineage client. This class is the entry point for all lineage related operations. An
+ * instance of this class can be obtained via {@link TachyonLineageFileSystem#get}.This class is
+ * thread safe.
+ */
+@PublicApi
+public class TachyonLineageFileSystem extends TachyonFileSystem {
+  private static final Logger LOG = LoggerFactory.getLogger(Constants.LOGGER_TYPE);
+
+  private static TachyonLineageFileSystem sTachyonFileSystem;
+  protected LineageContext mContext;
+
+  public static synchronized TachyonLineageFileSystem get() {
+    if (sTachyonFileSystem == null) {
+      sTachyonFileSystem = new TachyonLineageFileSystem();
+    }
+    return sTachyonFileSystem;
+  }
+
+  protected TachyonLineageFileSystem() {
```

Comment:
I know. But then other developer cannot extend this publicApi, because no public constructor in the super class. I'm fine with this change if we change the class to final.